### PR TITLE
[TTAHUB-1029] Show all AR's on goals & objectives page, show all topics on approved reports table

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Fixes the following bugs:
- Objectives on the RTR page seem to only show one AR under "Activity Reports" even though they have been included on multiple ARs.
- Topics for newly created objectives do not display in the approved reports table, only the "legacy" topics field

## How to test

Reuse an existing goal and objective (that already has a report) on a new activity report, then approve it. You will see topics in the approved report table for the objective you created. Look up that goal & objective in the RTR. You will see both reports.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1029

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/Constants.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ export const DATE_DISPLAY_FORMAT = 'MM/DD/YYYY';
 export const DATEPICKER_VALUE_FORMAT = 'YYYY-MM-DD';
 export const EARLIEST_INC_FILTER_DATE = moment('2020-08-31');
 
-const LOCAL_STORAGE_CACHE_NUMBER = '0.1';
+const LOCAL_STORAGE_CACHE_NUMBER = '0.2';
 export const LOCAL_STORAGE_DATA_KEY = (id) => `ar-form-data-${id}-${LOCAL_STORAGE_CACHE_NUMBER}`;
```
Comment (thewatermethod) [inline frontend/src/Constants.js:240]: Unrelated to this change specifically, we do want to increment this number given the report data object is being transformed and we don't want any unexpected errors during the transition.

File: frontend/src/__tests__/localStorageCleanup.js [hunk 2]
```diff
@@ -38,12 +38,12 @@ describe('localStorageCleanup', () => {
...
-        'ar-form-data-3-0.1',
-        'ar-additional-data-3-0.1',
-        'ar-can-edit-3-0.1',
+        'ar-form-data-2-0.2',
+        'ar-additional-data-2-0.2',
+        'ar-can-edit-2-0.2',
+        'ar-form-data-3-0.2',
+        'ar-additional-data-3-0.2',
+        'ar-can-edit-3-0.2',
       ];
```
Comment (thewatermethod) [inline frontend/src/__tests__/localStorageCleanup.js:47]: update test for last change

File: frontend/src/components/ActivityReportsTable/ReportRow.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ function ReportRow({
     displayId,
     activityRecipients,
     startDate,
-    topics,
+    sortedTopics: topics,
     lastSaved,
```
Comment (thewatermethod) [inline frontend/src/components/ActivityReportsTable/ReportRow.js:27]: turns out we had a sorted topics virtual field, who knew

File: frontend/src/components/GoalCards/ObjectiveCard.js [hunk 4]
```diff
@@ -128,7 +128,7 @@ function ObjectiveCard({
       </li>
       <li className="display-flex padding-bottom-05 flex-align-start">
         <span className="margin-right-3 minw-15">Reasons</span>
-        {reasons && displayReasonsList(reasons.sort())}
+        {reasons && displayReasonsList(reasons)}
```
Comment (thewatermethod) [inline frontend/src/components/GoalCards/ObjectiveCard.js:131]: sorting these on the backend

File: frontend/src/fetchers/activityReports.js [hunk 5]
```diff
@@ -49,16 +49,43 @@ export const getReport = async (reportId) => {
   return report.json();
 };
```
Comment (thewatermethod) [inline frontend/src/fetchers/activityReports.js:51]: Below, we take the topics from the API and filter them into the relevant report

File: frontend/src/components/ActivityReportsTable/ReportRow.js (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/ActivityReportsTable/ReportRow.js:27]: Should we be overwriting the sorted topics?

**Review by kryswisnaskas — APPROVED** [review]
Nice! The topics as well as multiple ARs on objectives are now displayed. 👍

**Pull request merged**